src/expandableCalendar/WeekCalendar/presenter.js

```diff
@@ -0,0 +1,187 @@
+import _ from 'lodash';
+import React from 'react';
+import {sameWeek} from '../../dateutils';
+const commons = require('../commons');
+import XDate from 'xdate';
+
+const UPDATE_SOURCES = commons.UPDATE_SOURCES;
+// must be a positive number
+const NUMBER_OF_PAGES = 2;
+
+class Presenter {
+  constructor(props) {
+    this.list = React.createRef();
+    this.props = props;
+    this._applyAndroidRtlFix = commons.isAndroid && commons.isRTL;
+    // On Android+RTL there's an initial scroll that cause issues
+    this._firstAndroidRTLScrollIgnored = !this._applyAndroidRtlFix;
+  }
+
+  updateWith = props => {
+    this.props = props;
+  };
+
+  scrollToIndex = animated => {
+    this.list.current.scrollToIndex({animated, index: NUMBER_OF_PAGES});
+  };
+
+  isSameWeek = (date, prevDate, firstDay) => {
+    return sameWeek(date, prevDate, firstDay);
+  };
+
+  // Events
+
+  onDayPressed = (context, value) => {
+    _.invoke(context, 'setDate', value.dateString, UPDATE_SOURCES.DAY_PRESS);
+  };
+
+  onScroll = ({context, updateState, x, page, items, width}) => {
```

Comment:
Now we have `onScroll` in both WeekCalendat/index and the presenter?

Reply:
Merge issue - fixed, onScroll on `index` uses presenter logic